File: abc154/c/php/c.php
<?php
$n = trim(fgets(STDIN));
$a = explode(" ", trim(fgets(STDIN)));

sort($a);

for($i = 0; $i < $n - 1; $i++){
  if($a[$i] === $a[$i + 1]){
    echo "NO\n";
    return;
  }
}
echo "YES\n";
?>

File: abc154/c/rust/c.rs
use std::io::{self, Read};

fn main(){
    let mut buf = String::new();
    std::io::stdin().read_to_string(&mut buf).unwrap();
    let mut iter = buf.split_whitespace();
    let n: usize = iter.next().unwrap().parse().unwrap();
    let mut a: Vec<i32> = iter.map(|x| x.parse().unwrap()).collect();

    a.sort();

    for i in 0 .. n - 1{
        if a[i] == a[i + 1]{
            println!("NO");
            return;
        }
    }
    println!("YES")
}
